domain-services/fiscal-event-service/src/main/java/org/egov/validator/FiscalEventValidator.java

```diff
@@ -173,28 +170,46 @@ private void validateFiscalEventAmountDetails(FiscalEventRequest fiscalEventRequ
                 allAmountDetails.addAll(amountDetails);
             }
 
-            Set<String> coaIds = new HashSet<>();
+            Set<String> coaCodes = new HashSet<>();
             for (Amount amount : allAmountDetails) {
                 if (amount.getAmount() == null /*|| amount.getAmount().compareTo(BigDecimal.ZERO)==0*/)
-                    errorMap.put("AMOUNT", "Amount is missing in request for coaId : " + amount.getCoaId());
-                if (StringUtils.isNotBlank(amount.getCoaId()))
-                    coaIds.add(amount.getCoaId());
-                else
-                    throw new CustomException("COA_ID", "COA id is missing in the request data");
+                    errorMap.put("AMOUNT", "Amount is missing in request for coaCode : " + amount.getCoaCode());
+                if (StringUtils.isNotBlank(amount.getCoaCode()))
+                    coaCodes.add(amount.getCoaCode());
+                else {
+                    errorMap.put("COA_CODE", "COA Code is missing in the request data");
+                    throw new CustomException(errorMap);
+                }
             }
 
-            List<String> responseCoaIds = coaUtil.getCOAIdsFromCOAService(requestHeader, coaIds, tenantId);
+            JsonNode jsonNode = coaUtil.getCOACodesFromCOAService(requestHeader, coaCodes, tenantId);
```

Comment:
Here, we are fetching coaDetails based on coaCodes, right? Write a better function name.